pkg/network/http/ebpf_gotls.go

```diff
@@ -286,9 +344,8 @@ func (p *GoTLSProgram) Stop() {
 	if p == nil {
 		return
 	}
-	// In the future, this should stop the new process listener.
-	p.detachHooks()
```

Comment:
why don't we remove the hooks?

Reply:
This `Stop()` method is part of `subprogram` interface and is called [here](https://github.com/DataDog/datadog-agent/blob/e85f52bbb754f16b7c4df35562d787198eee1590/pkg/network/http/ebpf_main.go#L268) as part of the exit procedure of `ebpfProgram` (which is also an `ebpfManager` as it embeds it). If you look just above the loop calling the subprograms `Stop()` methods, you'll see a call to the manager's `Stop()` method, which already does the work of detaching all probes. Hence why we do not need to do it here. This call to `detachHooks` was also causing the following kind of errors to be reported:
`[Error] failed detaching hook 0709b_6: the manager must be initialized first`
...which makes sense since the manager was stopped before.